src/cranbtree.c

```diff
@@ -205,6 +211,7 @@ void *cbt_update(cranbtree_t * bt, int key, void *object)
 	if (old_object == NULL)
 	{
```

Comment:
Note that the return value of this function is not an int but void* (a generic pointer). Thus, should either return NULL or a pointer to an object. Note how CBT_CLONE_BAD_OP is set and how the user of the library could find out about these errors later. **Hint: scan through the docs**